File: ruby/put_user_id_meta_data.rb
require 'oauth'
require 'json'

CONSUMER_KEY = "e4a55dbd46312950cf0d288ff42d74b4"
CONSUMER_SECRET = "b1f8c1c8388ef7c3cd4268ebfb733ba1"

consumer = OAuth::Consumer.new(
    CONSUMER_KEY,
    CONSUMER_SECRET,
    :site         => "https://api.howareyou.com",
)

# These values are required to decrypt data
# You will have to obtain a new session id and key from the authentication endpoint
headers = {
  "X-CHID-SESSION" => JSON.dump({
    "id"  => "DEzQvQE+O+JrgbZtf8+xK7IYuySQTfLMbpHHYx/wyzk=",
    "key" => "+NJmNrd4E5L6t/T37DZKauOGzlQT28ODko72J2T/Vgg=",
  }),
}

access_token = OAuth::AccessToken.new(consumer)

params = {"key" => "meta-key", "value" => "meta-value"}

response = access_token.put("/users/c5323558d6750f5571655b71ae2a2380/meta_data.json", params, headers)

p JSON.parse(response.body)

# =>
# {
#     "key"     => "meta_key",
#     "value"   => "meta_value",
#     "user_id" => "c5323558d6750f5571655b71ae2a2380",
# }

File: js/put_user_id_meta_data.js
(function () {
  var auth = ohauth.headerGenerator({
    consumer_key:    '6aa8d2199357db6670b03653f35a2e91',
    consumer_secret: '8fa254320524c355da88df808fc4c25e'
  });

  var url = 'https://api.howareyou.com/users/c5323558d6750f5571655b71ae2a2380/meta_data.json';

  var xhr = new XMLHttpRequest();

  xhr.onreadystatechange = function () {
    if (xhr.readyState == 4) {
      console.log( xhr.responseText );
    }
  }

  var params = { key:'meta-key', value:'meta-value' }

  xhr.open('PUT', url);
  xhr.setRequestHeader('Content-type','application/x-www-form-urlencoded');
  xhr.setRequestHeader('Authorization', auth('PUT', url, params));

  // These values are required to decrypt data
  // You will have to obtain a new session id and key from the authentication endpoint
  xhr.setRequestHeader('X-CHID-SESSION', JSON.stringify({
    "id":  "DMjIvrewGtwVKkD/UIVAUwrtrnNi81U7BSv8MR4YIBU=",
    "key": "h0058POWHRm95zMYMh8uEvdLDcoTYKOssdmQzwnc60c="
  }));

  xhr.send(ohauth.qsString(params));
})();

/*
 * {
 *     "key"     => "meta_key",
 *     "value"   => "meta_value",
 *     "user_id" => "c5323558d6750f5571655b71ae2a2380",
 * }
 */
